diff --git a/src/amiga/amiga.cpp b/src/amiga/amiga.cpp
index eb4337a..d6cf669 100644
--- a/src/amiga/amiga.cpp
+++ b/src/amiga/amiga.cpp
@@ -28,7 +28,8 @@
 #include <base/pathmanager.h>
 #include <QImage>
 #include <QSemaphore>
-#include <QApplication>
+#include <QGuiApplication>
+#include <QQuickView>
 
 static volatile bool amigaGoingShutdown = false;
 static QImage amigaFrame;
@@ -178,7 +179,9 @@ void AmigaEmu::sl() {
 int main(int argc, char *argv[]) {
 	if (argc < 2)
 		return -1;
-	QApplication app(argc, argv);
-	EmuView view(&amigaEmu, argv[1]);
-	return app.exec();
+    QGuiApplication *app = new QGuiApplication(argc, argv);
+    QQuickView *view = new QQuickView();
+
+	EmuView emuView(&amigaEmu, argv[1], view);
+	return app->exec();
 }
